internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -677,6 +731,12 @@ func resourceDatabricksWorkspaceRead(d *pluginsdk.ResourceData, meta interface{}
 		d.Set("managed_resource_group_id", model.Properties.ManagedResourceGroupId)
 		d.Set("managed_resource_group_name", managedResourceGroupID.ResourceGroup)
 
+		defaultStorageFirewall := model.Properties.DefaultStorageFirewall
+		if defaultStorageFirewall != nil {
```

Comment:
Minor, but can probably combine these

```suggestion
		if defaultStorageFirewall := model.Properties.DefaultStorageFirewall; defaultStorageFirewall != nil {
```